generator/mavgen_java.py

```diff
@@ -301,13 +369,24 @@ def generate_MAVLinkMessage(directory, xml_list):
     */
     public CRC crc;
 
-    public MAVLinkPacket(int payloadLength){
+    public MAVLinkPacket(int payloadLength) {
```

Comment:
There should be a `public MAVLinkPacket(int payloadLength, Protocol protocol)` constructor that the single arg one calls rather then calling `setProtocol(Protocol protocol)` all the time in the parser.